packages/containers/src/List/List.container.js

```diff
@@ -104,12 +114,49 @@ class List extends React.Component {
 		this.props.setState({ displayMode: payload });
 	}
 
+	onToggleMultiSelection(event, data) {
+		const state = this.props.state.toJS();
+		let selectedItems = [].concat(state.selectedItems);
+		if (selectedItems.find(itemKey => itemKey === data[this.props.multiSelectionKey])) {
+			selectedItems = selectedItems.filter(
```

Comment:
This is not really efficient. You go over the array to check if the item exists.
If it exists, you go over all elements of the array to filter.

Instead you can : 
* `slice()` to have a new array ref
* get the index of the element you want (`array.indexOf(itemKey)`)
* index > -1, you remove it (`splice()`)
* else you push it